src/app/wallets/check-limit-helpers.ts

```diff
@@ -1,43 +1,57 @@
-import { getTwoFALimits, getUserLimits, MS_PER_DAY } from "@config"
+import { getTwoFALimits, getAccountLimits, MS_PER_DAY } from "@config"
 import { LimitsChecker } from "@domain/accounts"
+import { toSats } from "@domain/bitcoin"
+import { toCents } from "@domain/fiat"
 import { TwoFA, TwoFANewCodeNeededError } from "@domain/twoFA"
+import { WalletCurrency } from "@domain/wallets"
 import { LedgerService } from "@services/ledger"
 
 export const checkIntraledgerLimits = async ({
   amount,
   walletId,
+  walletCurrency,
   account,
+  dCConverter,
 }: {
-  amount: Satoshis
+  amount: CurrencyBaseAmount
```

Comment:
Shouldn't this be USDCents?

Reply:
> `Satoshis`, `USDCents`, `DisplayCurrencyBase` and `CurrencyBaseAmount`?.

so `CurrencyBaseAmount` is not really an additional types as it is for the 3 others:

```
type CurrencyBaseAmount = Satoshis | UsdCents
```

This means that the value attached to it can either be a Satoshis or a UsdCents.

There are many scenario in which this arise. For instance, when we read an entry from the ledger, we can't say it's a `Satoshis` any more, it's currency dependent. so it should either be resolves to UsdCents or Satoshis later down where more context is available. 

It's the same logic with the Limit checkers. The amount passed can be either, and the function will decided based on the currency wallet which denomination should be used for the amount based on walletCurrency.

Does that make sense?

Reply:
I think I understand - but this is IMO the wrong way to model it. We should be using a generic for these situations:
```
<T extends Satoshis | UsdCents> someFunction({amount}: {amount: T})
```

Not saying we have to introduce that now in this PR - but when we do the `BigInt` refactoring this should be in scope IMO.

Reply:
> `<T extends Satoshis | UsdCents> someFunction({amount}: {amount: T})`

we don't necessarily know which type is it as some point in the case. at least I have designed it as is.

ie: you get an amount from graphql, but the type can be either `Satoshis` or `UsdCents`. I guess this could be types at the same time we look into the wallet currency type. might be a better design indeed.